mapping/mapping.go

```
@@ -0,0 +1,131 @@
+	attrs inventory.DeviceAttributes, update bool) (inventory.DeviceAttributes, error) {
+	var mapping *model.Mapping
+	var err error
+	if update {
+		mapping, err = m.updateAndGetMapping(ctx, tenantID, attrs)
+	} else {
+		mapping, err = m.getMapping(ctx, tenantID)
+	}
+	if err != nil {
+		return nil, err
+	}
+	attributesToFields := attributesToFields(mapping.Inventory)
+	return mapAttributes(attrs, attributesToFields), nil
+}
+
+// ReverseInventoryAttribute looks up the inventory attribute name from the ES field
+func (m *mapper) ReverseInventoryAttributes(ctx context.Context, tenantID string,
+	attrs inventory.DeviceAttributes) (inventory.DeviceAttributes, error) {
+	mapping, err := m.getMapping(ctx, tenantID)
+	if err != nil {
+		return nil, err
+	}
+	attributesToFields := fieldsToAttributes(mapping.Inventory)
+	return mapAttributes(attrs, attributesToFields), nil
+}
+
+func (m *mapper) getMapping(ctx context.Context, tenantID string) (*model.Mapping, error) {
+	return m.ds.GetMapping(ctx, tenantID)
+}
+
+func (m *mapper) updateAndGetMapping(ctx context.Context, tenantID string,
+	attrs inventory.DeviceAttributes) (*model.Mapping, error) {
+	inventoryMapping := make([]string, 0, len(attrs))
+	for i := 0; i < len(attrs); i++ {
+		if attrs[i].Scope == inventory.AttrScopeInventory {
+			inventoryMapping = append(inventoryMapping, attrs[i].Name)
+		}
+	}
+	mapping, err := m.ds.UpdateAndGetMapping(ctx, tenantID, inventoryMapping)
+	if err != nil {
+		return nil, err
+	}
+	return mapping, nil
+}
+
+func mapAttributes(attrs inventory.DeviceAttributes,
+	mapping map[string]string) inventory.DeviceAttributes {
+	mappedAttrs := make(inventory.DeviceAttributes, 0, len(attrs))
+	for i := 0; i < len(attrs); i++ {
+		var attrName string
+		if attrs[i].Scope != inventory.AttrScopeInventory {
+			attrName = attrs[i].Name
+		} else if name, ok := mapping[attrs[i].Name]; ok {
+			attrName = name
+		} else {
+			attrName = attrs[i].Name
```

Shouldn't this throw an error or be silently ignored? This can potentially allow users to index fields that are outside the fixed set of fields.

---

Done here: https://github.com/mendersoftware/reporting/pull/79/commits/7bbc1d08358af3dbe90bf1ace3f2c7c527d2f4fc